File: src/js/utils/clipboard.js
// Clipboard helpers — WebView blocks navigator.clipboard.readText on Linux/Tauri.

const { invoke } = window.__TAURI__?.core ?? {};

export async function clipboardReadText() {
    if (invoke) {
        try {
            return await invoke('clipboard_read_text');
        } catch {
            /* fall through */
        }
    }
    if (navigator.clipboard?.readText) {
        return navigator.clipboard.readText();
    }
    throw new Error('Clipboard read is not available');
}

export async function clipboardWriteText(text) {
    if (invoke) {
        try {
            await invoke('clipboard_write_text', { text });
            return;
        } catch {
            /* fall through */
        }
    }
    if (navigator.clipboard?.writeText) {
        await navigator.clipboard.writeText(text);
        return;
    }
    throw new Error('Clipboard write is not available');
}

File: src-tauri/src/commands/clipboard.rs
use arboard::Clipboard;

#[tauri::command]
pub fn clipboard_read_text() -> Result<String, String> {
    Clipboard::new()
        .map_err(|e| format!("Clipboard unavailable: {}", e))?
        .get_text()
        .map_err(|e| format!("Failed to read clipboard: {}", e))
}

#[tauri::command]
pub fn clipboard_write_text(text: String) -> Result<(), String> {
    Clipboard::new()
        .map_err(|e| format!("Clipboard unavailable: {}", e))?
        .set_text(text)
        .map_err(|e| format!("Failed to write clipboard: {}", e))
}
